pkg/kubelet/cm/devicemanager/endpoint.go

```diff
@@ -177,8 +177,8 @@ func dial(unixSocketPath string) (pluginapi.DevicePluginClient, *grpc.ClientConn
 	defer cancel()
 
 	c, err := grpc.DialContext(ctx, unixSocketPath, grpc.WithInsecure(), grpc.WithBlock(),
-		grpc.WithDialer(func(addr string, timeout time.Duration) (net.Conn, error) {
-			return net.DialTimeout("unix", addr, timeout)
+		grpc.WithContextDialer(func(ctx context.Context, addr string) (net.Conn, error) {
+			return (&net.Dialer{}).DialContext(ctx, "unix", addr)
```

Comment:
do we still need to call WithContextDialer (I expected this to be the default implementation... xref https://github.com/grpc/grpc-go/blob/master/clientconn.go#L198-L203)

Reply:
Looks to me that switching to `WithTimeout` and using the default dialer impl like @liggitt suggested should preserve the previous behavior.

Reply:
@liggitt  @jpbetz 
Maybe we can simple write like this :
`c, err := grpc.DialContext(ctx, unixSocketPath, grpc.WithInsecure(), grpc.WithBlock())`
Because 
1. `grpc.WithDialer` is actually set timeout via `context.deadline` , xref https://github.com/grpc/grpc-go/blob/788ffe62759ade61e0d6273e480c9e8f3d09d30d/dialoptions.go#L358-L360
2. `grpc.DialContext` finally call `dial.DialContext` which will set deadline from context. xref https://github.com/golang/go/blob/cf06b9aa81004a017e5c98422fce2fafd1f24921/src/net/dial.go#L368-L370
These two approach is the same, right ?
If so, we can safely skip this option.